### PR TITLE
Add procfs entry for /proc/[pid]/stat 

### PR DESCRIPTION
**Pull request by vtikoo**
### Summary
- Adds support for /proc/[pid]/support.
- Fixes pid specific entries - /proc/[pid]/maps and /proc/[pid]/status, when called from a different process.
- Adds lookup function for a process thread by pid. Returns pointer to process thread on success, NULL on failure -
```C
myst_thread_t* myst_find_process(pid_t pid);
```

### CODE REVIEW

File: kernel/procfs.c [hunk 1]
```diff
@@ -375,6 +402,120 @@ static int _status_vcallback(myst_buf_t* vbuf)
     return ret;
 }
 
+static char get_process_state(myst_thread_t* process_thread)
+{
+    if (process_thread->signal.waiting_on_event)
+    {
```
Comment (Francis-Liu) [inline kernel/procfs.c:408]: nit: remove {}

File: tests/procfs/Makefile [hunk 2]
```diff
@@ -25,4 +27,4 @@ myst:
 	$(MAKE) -C $(TOP)/tools/myst
 
 clean:
-	rm -rf $(APPDIR) rootfs export ramfs
+	rm -rf $(APPDIR) rootfs export ramfs *.o
```
Comment (Francis-Liu) [inline tests/procfs/Makefile:30]: Is export a file name?

**Review by Francis-Liu — APPROVED** [review]
Great work!

File: kernel/procfs.c (hunk 1 above)
Comment (vtikoo) [reply kernel/procfs.c]: Fixed.

File: tests/procfs/Makefile (hunk 2 above)
Comment (vtikoo) [reply tests/procfs/Makefile]: Thanks for catching this. This probably crept in at some point.

**Pull request merged**